packages/ember-runtime/lib/system/string.js

```diff
@@ -28,13 +28,24 @@ var CAMELIZE_CACHE = new Cache(1000, function(key) {
   });
 });
 
-var STRING_CLASSIFY_REGEXP_1 = (/(\-|\_|\.|\s)+(.)?/g);
-var STRING_CLASSIFY_REGEXP_2 = (/(^|\/|\.)([a-z])/g);
+var STRING_CLASSIFY_REGEXP_1 = (/^(\-)+(.)?/);
+var STRING_CLASSIFY_REGEXP_2 = (/(.)(\-|\_|\.|\s)+(.)?/g);
+var STRING_CLASSIFY_REGEXP_3 = (/(^|\/|\.)([a-z])/g);
 
 var CLASSIFY_CACHE = new Cache(1000, function(str) {
-  return str.replace(STRING_CLASSIFY_REGEXP_1, function(match, separator, chr) {
-    return chr ? chr.toUpperCase() : '';
-  }).replace(STRING_CLASSIFY_REGEXP_2, function(match, separator, chr) {
+  return str
+  .split('/')
+  .map(function(piece) {
```

Comment:
We can't use `Array.prototype.map` in release-1-13 (because it must support ES3 browsers), can you swap this to a `for` loop instead to allow this to be pulled into those branches?